File: kaltura/api_v3/generator/output/dotNet/KalturaClient/Types/KalturaBaseEntryFilter.cs
using System;
using System.Xml;
using System.Collections.Generic;

namespace Kaltura
{
	public class KalturaBaseEntryFilter : KalturaFilter
	{
		#region Properties
		public string IdEqual = null;
		public string IdIn = null;
		public string NameLike = null;
		public string NameMultiLikeOr = null;
		public string NameMultiLikeAnd = null;
		public string NameEqual = null;
		public int PartnerIdEqual = Int32.MinValue;
		public string PartnerIdIn = null;
		public string UserIdEqual = null;
		public string TagsLike = null;
		public string TagsMultiLikeOr = null;
		public string TagsMultiLikeAnd = null;
		public string AdminTagsLike = null;
		public string AdminTagsMultiLikeOr = null;
		public string AdminTagsMultiLikeAnd = null;
		public KalturaEntryStatus StatusEqual = (KalturaEntryStatus)Int32.MinValue;
		public string StatusIn = null;
		public KalturaEntryType TypeEqual = (KalturaEntryType)Int32.MinValue;
		public string TypeIn = null;
		public int CreatedAtGreaterThanOrEqual = Int32.MinValue;
		public int CreatedAtLessThanOrEqual = Int32.MinValue;
		public int GroupIdEqual = Int32.MinValue;
		public string SearchTextMatchAnd = null;
		public string SearchTextMatchOr = null;
		public string TagsNameMultiLikeOr = null;
		public string TagsAdminTagsMultiLikeOr = null;
		public string TagsAdminTagsNameMultiLikeOr = null;
		public string TagsNameMultiLikeAnd = null;
		public string TagsAdminTagsMultiLikeAnd = null;
		public string TagsAdminTagsNameMultiLikeAnd = null;
		#endregion

		#region CTor
		public KalturaBaseEntryFilter()
		{
		}

		public KalturaBaseEntryFilter(XmlElement node) : base(node)
		{
			foreach (XmlElement propertyNode in node.ChildNodes)
			{
				string txt = propertyNode.InnerText;
				switch (propertyNode.Name)
				{
					case "idEqual":
						this.IdEqual = txt;
						continue;
					case "idIn":
						this.IdIn = txt;
						continue;
					case "nameLike":
						this.NameLike = txt;
						continue;
					case "nameMultiLikeOr":
						this.NameMultiLikeOr = txt;
						continue;
					case "nameMultiLikeAnd":
						this.NameMultiLikeAnd = txt;
						continue;
					case "nameEqual":
						this.NameEqual = txt;
						continue;
					case "partnerIdEqual":
						this.PartnerIdEqual = ParseInt(txt);
						continue;
					case "partnerIdIn":
						this.PartnerIdIn = txt;
						continue;
					case "userIdEqual":
						this.UserIdEqual = txt;
						continue;
					case "tagsLike":
						this.TagsLike = txt;
						continue;
					case "tagsMultiLikeOr":
						this.TagsMultiLikeOr = txt;
						continue;
					case "tagsMultiLikeAnd":
						this.TagsMultiLikeAnd = txt;
						continue;
					case "adminTagsLike":
						this.AdminTagsLike = txt;
						continue;
					case "adminTagsMultiLikeOr":
						this.AdminTagsMultiLikeOr = txt;
						continue;
					case "adminTagsMultiLikeAnd":
						this.AdminTagsMultiLikeAnd = txt;
						continue;
					case "statusEqual":
						this.StatusEqual = (KalturaEntryStatus)ParseEnum(typeof(KalturaEntryStatus), txt);
						continue;
					case "statusIn":
						this.StatusIn = txt;
						continue;
					case "typeEqual":
						this.TypeEqual = (KalturaEntryType)ParseEnum(typeof(KalturaEntryType), txt);
						continue;
					case "typeIn":
						this.TypeIn = txt;
						continue;
					case "createdAtGreaterThanOrEqual":
						this.CreatedAtGreaterThanOrEqual = ParseInt(txt);
						continue;
					case "createdAtLessThanOrEqual":
						this.CreatedAtLessThanOrEqual = ParseInt(txt);
						continue;
					case "groupIdEqual":
						this.GroupIdEqual = ParseInt(txt);
						continue;
					case "searchTextMatchAnd":
						this.SearchTextMatchAnd = txt;
						continue;
					case "searchTextMatchOr":
						this.SearchTextMatchOr = txt;
						continue;
					case "tagsNameMultiLikeOr":
						this.TagsNameMultiLikeOr = txt;
						continue;
					case "tagsAdminTagsMultiLikeOr":
						this.TagsAdminTagsMultiLikeOr = txt;
						continue;
					case "tagsAdminTagsNameMultiLikeOr":
						this.TagsAdminTagsNameMultiLikeOr = txt;
						continue;
					case "tagsNameMultiLikeAnd":
						this.TagsNameMultiLikeAnd = txt;
						continue;
					case "tagsAdminTagsMultiLikeAnd":
						this.TagsAdminTagsMultiLikeAnd = txt;
						continue;
					case "tagsAdminTagsNameMultiLikeAnd":
						this.TagsAdminTagsNameMultiLikeAnd = txt;
						continue;
				}
			}
		}
		#endregion

		#region Methods
		public new KalturaParams ToParams()
		{
			KalturaParams kparams = base.ToParams();
			kparams.AddStringIfNotNull("idEqual", this.IdEqual);
			kparams.AddStringIfNotNull("idIn", this.IdIn);
			kparams.AddStringIfNotNull("nameLike", this.NameLike);
			kparams.AddStringIfNotNull("nameMultiLikeOr", this.NameMultiLikeOr);
			kparams.AddStringIfNotNull("nameMultiLikeAnd", this.NameMultiLikeAnd);
			kparams.AddStringIfNotNull("nameEqual", this.NameEqual);
			kparams.AddIntIfNotNull("partnerIdEqual", this.PartnerIdEqual);
			kparams.AddStringIfNotNull("partnerIdIn", this.PartnerIdIn);
			kparams.AddStringIfNotNull("userIdEqual", this.UserIdEqual);
			kparams.AddStringIfNotNull("tagsLike", this.TagsLike);
			kparams.AddStringIfNotNull("tagsMultiLikeOr", this.TagsMultiLikeOr);
			kparams.AddStringIfNotNull("tagsMultiLikeAnd", this.TagsMultiLikeAnd);
			kparams.AddStringIfNotNull("adminTagsLike", this.AdminTagsLike);
			kparams.AddStringIfNotNull("adminTagsMultiLikeOr", this.AdminTagsMultiLikeOr);
			kparams.AddStringIfNotNull("adminTagsMultiLikeAnd", this.AdminTagsMultiLikeAnd);
			kparams.AddEnumIfNotNull("statusEqual", this.StatusEqual);
			kparams.AddStringIfNotNull("statusIn", this.StatusIn);
			kparams.AddEnumIfNotNull("typeEqual", this.TypeEqual);
			kparams.AddStringIfNotNull("typeIn", this.TypeIn);
			kparams.AddIntIfNotNull("createdAtGreaterThanOrEqual", this.CreatedAtGreaterThanOrEqual);
			kparams.AddIntIfNotNull("createdAtLessThanOrEqual", this.CreatedAtLessThanOrEqual);
			kparams.AddIntIfNotNull("groupIdEqual", this.GroupIdEqual);
			kparams.AddStringIfNotNull("searchTextMatchAnd", this.SearchTextMatchAnd);
			kparams.AddStringIfNotNull("searchTextMatchOr", this.SearchTextMatchOr);
			kparams.AddStringIfNotNull("tagsNameMultiLikeOr", this.TagsNameMultiLikeOr);
			kparams.AddStringIfNotNull("tagsAdminTagsMultiLikeOr", this.TagsAdminTagsMultiLikeOr);
			kparams.AddStringIfNotNull("tagsAdminTagsNameMultiLikeOr", this.TagsAdminTagsNameMultiLikeOr);
			kparams.AddStringIfNotNull("tagsNameMultiLikeAnd", this.TagsNameMultiLikeAnd);
			kparams.AddStringIfNotNull("tagsAdminTagsMultiLikeAnd", this.TagsAdminTagsMultiLikeAnd);
			kparams.AddStringIfNotNull("tagsAdminTagsNameMultiLikeAnd", this.TagsAdminTagsNameMultiLikeAnd);
			return kparams;
		}
		#endregion
	}
}

File: kaltura/api_v3/lib/types/filters/KalturaBaseEntryFilter.php
<?php
/**
 * @package api
 * @subpackage filters
 */
class KalturaBaseEntryFilter extends KalturaFilter
{
	private $map_between_objects = array
	(
		"idEqual" => "_eq_id",
		"idIn" => "_in_id",
		"nameLike" => "_like_name",
		"nameMultiLikeOr" => "_mlikeor_name",
		"nameMultiLikeAnd" => "_mlikeand_name",
		"nameEqual" => "_eq_name",
		"partnerIdEqual" => "_eq_partner_id",
		"partnerIdIn" => "_in_partner_id",
		"userIdEqual" => "_eq_user_id",
		"tagsLike" => "_like_tags",
		"tagsMultiLikeOr" => "_mlikeor_tags",
		"tagsMultiLikeAnd" => "_mlikeand_tags",
		"adminTagsLike" => "_like_admin_tags",
		"adminTagsMultiLikeOr" => "_mlikeor_admin_tags",
		"adminTagsMultiLikeAnd" => "_mlikeand_admin_tags",
		"statusEqual" => "_eq_status",
		"statusIn" => "_in_status",
		"typeEqual" => "_eq_type",
		"typeIn" => "_in_type",
		"createdAtGreaterThanOrEqual" => "_gte_created_at",
		"createdAtLessThanOrEqual" => "_lte_created_at",
		"groupIdEqual" => "_eq_group_id",
		"searchTextMatchAnd" => "_matchand_search_text",
		"searchTextMatchOr" => "_matchor_search_text",
		"tagsNameMultiLikeOr" => "_mlikeor_tags-name",
		"tagsAdminTagsMultiLikeOr" => "_mlikeor_tags-adminTags",
		"tagsAdminTagsNameMultiLikeOr" => "_mlikeor_tags-adminTags-name",
		"tagsNameMultiLikeAnd" => "_mlikeand_tags-name",
		"tagsAdminTagsMultiLikeAnd" => "_mlikeand_tags-adminTags",
		"tagsAdminTagsNameMultiLikeAnd" => "_mlikeand_tags-adminTags-name",
	);

	private $order_by_map = array
	(
		"+name" => "+name",
		"-name" => "-name",
		"+createdAt" => "+created_at",
		"-createdAt" => "-created_at",
		"+rank" => "+rank",
		"-rank" => "-rank",
	);

	public function getMapBetweenObjects()
	{
		return array_merge(parent::getMapBetweenObjects(), $this->map_between_objects);
	}

	public function getOrderByMap()
	{
		return array_merge(parent::getOrderByMap(), $this->order_by_map);
	}

	/**
	 * This filter should be in use for retrieving only a specific entry (identified by its entryId).
	 * 
	 * @var string
	 */
	public $idEqual;

	/**
	 * This filter should be in use for retrieving few specific entries (string should include comma separated list of entryId strings).
	 * 
	 * @var string
	 */
	public $idIn;

	/**
	 * This filter should be in use for retrieving specific entries while applying an SQL 'LIKE' pattern matching on entry names. It should include only one pattern for matching entry names against.
	 * 
	 * @var string
	 */
	public $nameLike;

	/**
	 * This filter should be in use for retrieving specific entries, while applying an SQL 'LIKE' pattern matching on entry names. It could include few (comma separated) patterns for matching entry names against, while applying an OR logic to retrieve entries that match at least one input pattern.
	 * 
	 * @var string
	 */
	public $nameMultiLikeOr;

	/**
	 * This filter should be in use for retrieving specific entries, while applying an SQL 'LIKE' pattern matching on entry names. It could include few (comma separated) patterns for matching entry names against, while applying an AND logic to retrieve entries that match all input patterns.
	 * 
	 * @var string
	 */
	public $nameMultiLikeAnd;

	/**
	 * This filter should be in use for retrieving entries with a specific name.
	 * 
	 * @var string
	 */
	public $nameEqual;

	/**
	 * This filter should be in use for retrieving only entries which were uploaded by/assigned to users of a specific Kaltura Partner (identified by Partner ID).
	 * 
	 * @var int
	 */
	public $partnerIdEqual;

	/**
	 * This filter should be in use for retrieving only entries within Kaltura network which were uploaded by/assigned to users of few Kaltura Partners  (string should include comma separated list of PartnerIDs)
	 * 
	 * @var string
	 */
	public $partnerIdIn;

	/**
	 * This filter parameter should be in use for retrieving only entries, uploaded by/assigned to a specific user (identified by user Id).
	 * 
	 * @var string
	 */
	public $userIdEqual;

	/**
	 * This filter should be in use for retrieving specific entries while applying an SQL 'LIKE' pattern matching on entry tags. It should include only one pattern for matching entry tags against.
	 * 
	 * @var string
	 */
	public $tagsLike;

	/**
	 * This filter should be in use for retrieving specific entries, while applying an SQL 'LIKE' pattern matching on tags.  It could include few (comma separated) patterns for matching entry tags against, while applying an OR logic to retrieve entries that match at least one input pattern.
	 * 
	 * @var string
	 */
	public $tagsMultiLikeOr;

	/**
	 * This filter should be in use for retrieving specific entries, while applying an SQL 'LIKE' pattern matching on tags.  It could include few (comma separated) patterns for matching entry tags against, while applying an AND logic to retrieve entries that match all input patterns.
	 * 
	 * @var string
	 */
	public $tagsMultiLikeAnd;

	/**
	 * This filter should be in use for retrieving specific entries while applying an SQL 'LIKE' pattern matching on entry tags, set by an ADMIN user. It should include only one pattern for matching entry tags against.
	 * 
	 * @var string
	 */
	public $adminTagsLike;

	/**
	 * This filter should be in use for retrieving specific entries, while applying an SQL 'LIKE' pattern matching on tags, set by an ADMIN user.  It could include few (comma separated) patterns for matching entry tags against, while applying an OR logic to retrieve entries that match at least one input pattern.
	 * 
	 * @var string
	 */
	public $adminTagsMultiLikeOr;

	/**
	 * This filter should be in use for retrieving specific entries, while applying an SQL 'LIKE' pattern matching on tags, set by an ADMIN user.  It could include few (comma separated) patterns for matching entry tags against, while applying an OR logic to retrieve entries that match all input patterns.
	 * 
	 * @var string
	 */
	public $adminTagsMultiLikeAnd;

	/**
	 * This filter should be in use for retrieving only entries, at a specific {@link ?object=KalturaEntryStatus KalturaEntryStatus}.
	 * 
	 * @var KalturaEntryStatus
	 */
	public $statusEqual;

	/**
	 * This filter should be in use for retrieving only entries, at few specific {@link ?object=KalturaEntryStatus KalturaEntryStatus}.
	 * 
	 * @var string
	 */
	public $statusIn;

	/**
	 * 
	 * 
	 * @var KalturaEntryType
	 */
	public $typeEqual;

	/**
	 * This filter should be in use for retrieving entries of few {@link ?object=KalturaEntryType KalturaEntryType} (string should include a comma separated list of {@link ?object=KalturaEntryType KalturaEntryType} enumerated parameters).
	 * 
	 * @var string
	 */
	public $typeIn;

	/**
	 * This filter parameter should be in use for retrieving only entries which were created at Kaltura system after a specific time/date (standard timestamp format).
	 * 
	 * @var int
	 */
	public $createdAtGreaterThanOrEqual;

	/**
	 * This filter parameter should be in use for retrieving only entries which were created at Kaltura system before a specific time/date (standard timestamp format).
	 * 
	 * @var int
	 */
	public $createdAtLessThanOrEqual;

	/**
	 * 
	 * 
	 * @var int
	 */
	public $groupIdEqual;

	/**
	 * This filter should be in use for retrieving specific entries while search match the input string within all of the following metadata attributes: name, description, tags, adminTags.
	 * 
	 * @var string
	 */
	public $searchTextMatchAnd;

	/**
	 * This filter should be in use for retrieving specific entries while search match the input string within at least one of the following metadata attributes: name, description, tags, adminTags.
	 * 
	 * @var string
	 */
	public $searchTextMatchOr;

	/**
	 * @var string
	 */
	public $tagsNameMultiLikeOr;

	/**
	 * @var string
	 */
	public $tagsAdminTagsMultiLikeOr;

	/**
	 * @var string
	 */
	public $tagsAdminTagsNameMultiLikeOr;

	/**
	 * @var string
	 */
	public $tagsNameMultiLikeAnd;

	/**
	 * @var string
	 */
	public $tagsAdminTagsMultiLikeAnd;

	/**
	 * @var string
	 */
	public $tagsAdminTagsNameMultiLikeAnd;
}
